packages/twenty-server/src/engine/metadata-modules/remote-server/utils/validate-remote-server-type.util.ts

```diff
@@ -0,0 +1,38 @@
+import { BadRequestException } from '@nestjs/common';
+
+import { Repository } from 'typeorm';
+
+import {
+  FeatureFlagEntity,
+  FeatureFlagKeys,
+} from 'src/engine/core-modules/feature-flag/feature-flag.entity';
+import { RemoteServerType } from 'src/engine/metadata-modules/remote-server/remote-server.entity';
+
+export const validateRemoteServerType = async (
+  remoteServerType: RemoteServerType,
+  featureFlagRepository: Repository<FeatureFlagEntity>,
+  workspaceId: string,
+) => {
+  const featureFlagKey = getFeatureFlagKey(remoteServerType);
+
+  const featureFlag = await featureFlagRepository.findOneBy({
+    workspaceId,
+    key: featureFlagKey,
+    value: true,
+  });
+
+  const featureFlagEnabled = featureFlag && featureFlag.value;
+
+  if (!featureFlagEnabled) {
+    throw new BadRequestException('PostgreSQL integration is not enabled');
+  }
+};
+
+const getFeatureFlagKey = (remoteServerType: RemoteServerType) => {
+  switch (remoteServerType) {
+    case RemoteServerType.POSTGRES_FDW:
+      return FeatureFlagKeys.IsPostgreSQLIntegrationEnabled;
+    default:
+      throw new BadRequestException('PostgreSQL integration is not enabled');
```

Comment:
I don't think this is the right error message here. Should probably be `${remoteServerType} integration is not supported` or something, you check postgresql integration above already